### ISSUE TITLE
Download counts for packages and versions of packages

### ISSUE THREAD

**Issue by designfrontier** [rlidwka/sinopia#167]
Is this on the road map, for future updates to the web side? If not would you be interested in having it contributed as a feature?

**Comment by Dev661** [comment]
@designfrontier , @tcort Could you please elabortae how you would use Morgan for these metrics (lets say count of download,...)?

Your help is appreciated!

**Comment by tcort** [comment]
@Dev661 My idea was to use [morgan](https://github.com/expressjs/morgan) to log accesses in a standard web server access log format such as `combined`, `common`, ... ([example](https://github.com/expressjs/morgan#log-file-rotation)) and then post process the logs with an existing tool such as [awstats](http://www.awstats.org/) ([example download count output](http://www.nltechno.com/awstats/awstats.pl?config=destailleur.fr&framename=mainright&output=downloads)). My thinking was to use some standard formats and existing external tools in order to reduce complexity on the sinopia side (setting up morgan is just a few lines of code).

The comments following my comment make me think this isn't exactly what @designfrontier had in mind. It looks like he would rather have something built into sinopia to track and record downloads with a way to display and graph the information. This is totally doable but it's a lot of code to maintain and complexity compared to logging the requests to an access log.

**Comment by designfrontier** [comment]
@Dev661 Yeah I think the morgan solution may have merit... but it doesn't provide a simple way to get those metrics into the UI/API. It would externalize that information and that information would be as ephemeral as your log data is.

Would be preferable to have the counts logged to the actual package somehow. Perhaps in the JSON store as I mentioned above...
